<!-- lpetrich 27/06/18 -->

<launch>

    <group ns="cam1">
        <node pkg="usb_cam" type="usb_cam_node" name="camera" output="screen">
            <param name="video_device" value="/dev/video0" />
            <param name="framerate" value="15" />
        </node>
        <node name="trackers" pkg="mtf_bridge" type="vs_tracker_node" output="screen"> 
            <param name="shm_name" value="shm1"/>
            <remap from="/init_buffer" to="/cam1/init_buffer"/>
            <remap from="/input_image" to="/cam1/input_image"/>
            <remap from="/task_coordinates" to="/cam1/task_coordinates"/>
        </node>
        <node name="SharedImageWriter" pkg="mtf_bridge" type="SharedImageWriter" output="screen"> 
            <param name="shm_name" value="shm1"/>
            <remap from="/camera/image_raw" to="/cam1/camera/image_raw"/>
        </node>
    </group> 

    <group ns="cam2">
        <node pkg="usb_cam" type="usb_cam_node" name="camera" output="screen">
            <param name="video_device" value="/dev/video1" />
            <param name="framerate" value="15" />
        </node>
        <node name="trackers" pkg="mtf_bridge" type="vs_tracker_node" output="screen"> 
            <param name="shm_name" value="shm1"/>
            <remap from="/init_buffer" to="/cam1/init_buffer"/>
            <remap from="/input_image" to="/cam1/input_image"/>
            <remap from="/task_coordinates" to="/cam2/task_coordinates"/>
        </node>
        <node name="SharedImageWriter" pkg="mtf_bridge" type="SharedImageWriter" output="screen"> 
            <param name="shm_name" value="shm1"/>
            <remap from="/camera/image_raw" to="/cam2/camera/image_raw"/>
        </node>
    </group> 
    
    <node pkg="visual_servoing_gui" type="visual_servoing_gui" name="visual_servoing_gui" output="screen"/> 
    <node pkg="uncalibrated_visual_servoing" type="error_calculation" name="error_calculation" output="screen"/> 
<!--
    <node pkg="uncalibrated_visual_servoing" type="uvs" name="uvs" output="screen"/>
-->
</launch>